Project/Source/FileSystem/ModelImporter.cpp

```diff
@@ -696,12 +699,95 @@ bool ModelImporter::ImportModel(const char* filePath, JsonValue jMeta) {
 	// Load animations
 	if (assimpScene->mNumAnimations > 0) {
 		LOG("Importing animations");
-		std::vector<ResourceAnimation*> animations;
-		ComponentAnimation* animationComponent = root->GetChildren()[0]->CreateComponent<ComponentAnimation>();
+
+		JsonValue jResource = jResources[resourceIndex];
+		UID id = jResource[JSON_TAG_ID];
+		ResourceStateMachine* resourceStateMachine = App->resources->CreateResource<ResourceStateMachine>(filePath, id ? id : GenerateUID());
+		UID uidSM = resourceStateMachine->GetId();
+
+		jResource[JSON_TAG_TYPE] = GetResourceTypeName(resourceStateMachine->GetType());
+		jResource[JSON_TAG_ID] = resourceStateMachine->GetId();
+		resourceIndex += 1;
+
+		std::string resourceName = "animation";
+		std::string stateName = "state";
+		std::string clipName = "clip";
+
+		//ResourceAnimation* testAnim = nullptr; //TODO::Delete this
+
+		State initialState;
 		for (unsigned int i = 0; i < assimpScene->mNumAnimations; ++i) {
-			animationComponent->animationController.animationID = ImportAnimation(filePath, jMeta, assimpScene->mAnimations[i], assimpScene, resourceIndex)->GetId();
+			std::string parsedI = std::to_string(i);
+
+			ResourceAnimation* animation = ImportAnimation(filePath, jMeta, assimpScene->mAnimations[i], assimpScene, resourceIndex);
+
+			/*if (testAnim == nullptr) { //TODO::Delete this
+				testAnim = animation;
+			}*/
+
+			resourceStateMachine->resourceAnimations.insert(std::make_pair(resourceName + parsedI, animation));
+
+			JsonValue jResourceClip = jResources[resourceIndex];
+			UID idClip = jResourceClip[JSON_TAG_ID];
+			ResourceClip* clip = App->resources->CreateResource<ResourceClip>(filePath, idClip ? idClip : GenerateUID());
+			clip->Init(clipName + parsedI, animation->GetId(), 0, animation->keyFrames.size() - 1, true);
+			//clip.SetSpeed(2.0);
+
+			jResourceClip[JSON_TAG_TYPE] = GetResourceTypeName(clip->GetType());
+			jResourceClip[JSON_TAG_ID] = clip->GetId();
+			resourceIndex += 1;
+			clip->SaveToFile(clip->GetResourceFilePath().c_str());
+
+			//Adding initial state
+			initialState = resourceStateMachine->AddState(stateName + parsedI, clip->GetId());
 		}
-		// TODO: Improve for multiple animations
+
+		// TODO:: Hardcoded for testing animation
+		//Setting machine state
+		//Dead animation
+		/*std::string sState2 = "State2";
+		std::string clipName2 = "testClip2";
+		JsonValue jResourceClip2 = jResources[resourceIndex];
+		UID idClip2 = jResourceClip2[JSON_TAG_ID];
```

Comment:
Actually remove the code. We will always have it available in the previous commits in the branch

Reply:
We have this in case we need to test the animation in cases that it is breaks , we want to delete this once it has passed the vertical slide. But if needed we could delete and retrieve this from the commits

Reply:
Seems fine, Jaime also confirms this, so ignore my comment